packages/ipfs-core/src/components/bitswap/wantlist-for-peer.js

```diff
@@ -5,24 +5,26 @@ const { withTimeoutOption } = require('../../utils')
 
 /**
  * @param {Object} config
- * @param {import('..').IPFSBitSwap} config.bitswap
+ * @param {import('.').NetworkService} config.network
  */
-module.exports = ({ bitswap }) => {
+module.exports = ({ network }) => {
   /**
    * Returns the wantlist for a connected peer
    *
-   * @param {PeerId | CID | string | Uint8Array} peerId - A peer ID to return the wantlist for\
-   * @param {AbortOptions} [options]
-   * @returns {Promise<CID[]>} - An array of CIDs currently in the wantlist
-   *
    * @example
    * ```js
    * const list = await ipfs.bitswap.wantlistForPeer(peerId)
    * console.log(list)
    * // [ CID('QmHash') ]
    * ```
+   *
+   * @param {PeerId | CID | string | Uint8Array} peerId - A peer ID to return the wantlist for\
```

Comment:
I think better alternative would be accept `PeerId` here and let the user of the API do the conversion. That said if we were to add a type to `peer-id` I think it should be `ToPeerId` and there should be `PeerId.from(thing:ToPeerId):PeerId` function.

I think we already have couple of places with `ToThing` pattern.

Either way can we deal with trickling types into separate packages as a followup work ?

Reply:
Created a followup here https://github.com/libp2p/js-peer-id/issues/133

Reply:
We already have several `createFrom*` functions in `peer-id`, but they are targeted to a specific type https://github.com/libp2p/js-peer-id#import

I think this would be nice to have, I would just go with `PeerId.createFrom(param: ToPeerId)` to be consistent with what we already have.